### PR TITLE
chore(azure): Fix CIS 2.1 mapping

### LINKED ISSUE #4754: cis_2.1_azure incorrect checks resulting in incorrect outputs
### Steps to Reproduce

Running cis_2.1_azure compliancy checks results in incorrect results for controls 9.5, 9.6, 9.7, 9.8 and 9.9

### Expected behavior

PHP should use PHP check, Java should use Java check, etc.

### Actual Result with Screenshots or Logs

image

### How did you install Prowler?

Cloning the repository from github.com (git clone)

### Environment Resource

VM

### OS used

Kali Linux

### Prowler version

cis 2.1

### Pip version

-

### Context

_No response_

### PR DESCRIPTION
**Pull request by puchy22**
### Context

There are discrepancies in the mapping of certain checks within the Azure CIS 2.1 benchmark, as highlighted by a GitHub user in a reported [issue](https://github.com/prowler-cloud/prowler/issues/4754). Additionally, upon further inspection, I identified additional mappings that appear to be incorrect.

Fix #4754 

### Description

- [x] All mapping issues identified by the GitHub user are reviewed and corrected.
- [x] Additional incorrect mappings discovered during the review are also corrected.

### Checklist

- Are there new checks included in this PR? No
    - If so, do we need to update permissions for the provider? No
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/compliance/azure/cis_2.1_azure.json [hunk 1]
```diff
@@ -3213,66 +3213,6 @@
           "References": "https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-group-lock-resources:https://docs.microsoft.com/en-us/azure/azure-resource-manager/resource-manager-subscription-governance#azure-resource-locks:https://docs.microsoft.com/en-us/azure/governance/blueprints/concepts/resource-locking:https://learn.microsoft.com/en-us/security/benchmark/azure/mcsb-asset-management#am-4-limit-access-to-asset-management"
         }
       ]
-    },
```
Comment (MrCloudSec) [inline prowler/compliance/azure/cis_2.1_azure.json:3216]: Why are you removing this controls if they exist in CIS 2.1?

**Review by MrCloudSec — DISMISSED** [review]
Could you review this entire framework with the help of @pedrooot, please?

File: prowler/compliance/azure/cis_2.1_azure.json (hunk 1 above)
Comment (pedrooot) [reply prowler/compliance/azure/cis_2.1_azure.json:3216]: these were duplicated

**Review by pedrooot — APPROVED** [review]
🔝 🥇 Thanks for this

**Pull request merged**